### ISSUE TITLE
Key up/down event and Html select list issue

### ISSUE THREAD

**Issue by aitiu** [indiana-university/rivet-collapsible#30]
When up/down key is pressed on dropdown select list, the focus changes to collapsible panel instead of changing value in select list.

**Comment by levimcg** [comment]
Thanks for opening this issue. Would you be able to provide a little more detail about the issue you're having? It would help us out a lot if you could give us a link, or at least some code snippets, or screenshots.

Also can you confirm which version of Rivet itself (not the Collapsible add-on) you are using? This sort of sounds like this issue we were seeing a while back:

https://github.com/indiana-university/rivet-source/issues/53

Which was fixed in this pull request:

https://github.com/indiana-university/rivet-source/pull/66

and was released in Rivet `1.3.0`.

Thank you! 🙌

**Comment by aitiu** [comment]
We are using Rivet 1.5.0 and rivet-collapsible

![image](https://user-images.githubusercontent.com/19332118/59045358-ebb6dd00-884d-11e9-91a5-b0944a411326.png)

**Comment by levimcg** [comment]
@aitiu Great, thank you! I've pulled this into the backlog. We'll do a little more digging and let you know if we have any other questions.

Real quick, are you using the collapsible here in the _accordion_ configuration with the `data-accordion` attribute added to each? I'm thinking that might be where the issue is happening because there are some event handlers that deal with keyboard navigation that are most likely causing the issue. Thanks!

**Comment by levimcg** [comment]
I was able to dig into this a bit on Friday. I've got a test case set up here that i think isolates the bug.

Based on the screenshot above, I've added an input group with a `select` element and a `button` inside. The bug described above is happening when the input group (select + button) is used inside a collapsible in the accordion configurations I.e. a group of collapsible wrapped in an element with the `data-accordion` attribute.

```html
<div class="rvt-collapsible-accordion" data-accordion>
  <div class="rvt-collapsible">
    ... Collapsible heading, etc.
    <div class="rvt-collapsible__content">
      ... Input group with button
    </div>
  </div>
</div>
```

When you are focused on the `<button>` inside the input group and press the down key, it moves focus to the button element inside of the `rvt-collapsible__title` element.

The `data-accordion` attribute an event listener that handles keyboard navigation requirements for accordions (up and down arrow keys). I'm almost certain it has something to do with that. It might just be a matter of checking the event target inside the `handleKeyup` [event handler here on this line](https://github.com/indiana-university/rivet-collapsible/blob/develop/src/js/rivet-collapsible.js#L121).

**Comment by levimcg** [comment]
Guess I should include a link to the demo 🙃. I've also updated the original comment to include the link.

https://codepen.io/levimcg/pen/WBqrjo

**Comment by levimcg** [comment]
@aitiu We're working on this bug now, and just wanted to confirm a couple of things.

> when a down arrow key is pressed while the selected dropdownlist is highlighted, the focus moves to collapsible panel header (Catering Queue).

We we're able to reproduce a similar bug when pressing a down key while focused on the button element inside of an input group with a `select` element and a `button`. So with markup like this:

```html
<label for="go-2" class="rvt-sr-only">Search</label>
<div class="rvt-input-group">
  <select name="" id="go-2" class="rvt-input-group__input">
    <option value="">Option one</option>
    <option value="">Option two</option>
    <option value="">Option three</option>
    <option value="">Option four</option>
  </select>
  <div class="rvt-input-group__append">
      <button class="rvt-button">Go</button>
  </div>
</div>
```

We're able to reproduce the bug while pressing the down arrow key while focused on the **button**, but not while focused on the `<select>` element. See the following screenshot with the button focused:

![image](https://user-images.githubusercontent.com/5590034/59704532-5bbc5000-91ca-11e9-8e39-f7728d8ddc49.png)

When I press the down key in this state (focused button) it will move focus to the collapsible header above that. Can you confirm that focus is moving when you're focused on the select/dropdown element and NOT the button element?

Would you be able to give us access to your app so we can look at the markup?

Thanks for your help.

**Comment by aitiu** [comment]
Focus moves to header for both select/dropdown and button element when down arrow key is pressed

**Comment by illusivesunrae** [comment]
Hey, @aitiu, would it be possible for you to send us a link to your markup or paste a copy of it here? We are unable to reproduce the bug with the select/dropdown element.

**Comment by illusivesunrae** [comment]
@aitiu Just to follow up, Levi has investigated this in IE11 and managed to replicate the issue. He tried the fix in PR #31 and was successful. We will be merging the fix and creating a release this afternoon. I will report back when it is ready.

**Comment by illusivesunrae** [comment]
Alright, the fix for this should now be live with PR #32/ release 0.2.10. Feel free to reopen this issue if you are still experiencing issues following the release.
